test(agentic-ai): add e2e tests for A2A client connectors

mdii opened the pull request:
## Description

This PR adds end-to-end tests for the A2A client connectors operating in both tool mode and standalone mode. 

## Related issues

<!-- Which issues are closed by this PR or are related -->

closes #5503 

## Checklist

- [x] PR has a **milestone** or the `no milestone` label.
- [ ] Backport labels are added if these code changes should be backported. No backport label is added to the latest release, as this branch will be rebased onto main before the next release.

mdii commented on connectors-e2e-test/connectors-e2e-test-agentic-ai/src/test/java/io/camunda/connector/e2e/agenticai/a2a/A2aStandaloneTests.java:
```diff
@@ -0,0 +1,392 @@
+    var token = "test-token-123";
+    BpmnModelInstance bpmnModel =
+        BpmnFile.replace(
+            testProcess.getFile(), replace("id=\"A2A_Standalone\"", "id=\"A2A_Standalone_Token\""));
```
So that the connector runtime replaces the already registered A2A webhook with a new one for this process.

maff commented on connectors-e2e-test/connectors-e2e-test-agentic-ai/src/test/java/io/camunda/connector/e2e/agenticai/aiagent/langchain4j/outboundconnector/BaseL4JAiAgentConnectorTest.java:
```diff
@@ -92,11 +94,13 @@ protected ZeebeTest testBasicExecutionWithoutFeedbackLoop(
       Resource process,
       Function<ElementTemplate, ElementTemplate> elementTemplateModifier,
       String responseText,
+      Map<String, Object> extraProcessVariables,
```
🔧 Nit: would it make sense to move this above the `responseText`, so the parameters are logically ordered by input to start the process (process, modified template, extra variables) and then the expected response?

maff commented on connectors-e2e-test/connectors-e2e-test-agentic-ai/src/test/java/io/camunda/connector/e2e/agenticai/a2a/A2aStandaloneTests.java:
```diff
@@ -0,0 +1,392 @@
+    callWebhookEndpointWithDelay(webhookUrl, payload, Map.of(), delayInSeconds);
+  }
+
+  private void callWebhookEndpointWithDelay(
```
🔧 Is this the same logic as in the A2aIntegrationTestSupport, but with headers? Could we maybe consolidate those 2?

maff reviewed (commented):
A couple of remarks/smaller comments - overall amazing work! 💪

mdii commented on connectors-e2e-test/connectors-e2e-test-agentic-ai/src/test/java/io/camunda/connector/e2e/agenticai/a2a/A2aStandaloneTests.java:
```diff
@@ -0,0 +1,392 @@
+                """
+            <zeebe:property name="inbound.auth.type" value="BASIC" />
+            <zeebe:property name="inbound.auth.username" value="user1" />
+            <zeebe:property name="inbound.auth.password" value="pass123" />"""));
```
I will try :+1: 
I went with the easiest way but it could be brittle.

mdii commented:
@maff I addressed your comments. The tests are a bit faster now:

<img width="498" height="279" alt="image" src="https://github.com/user-attachments/assets/091e9546-b7ef-41db-8d5f-b717a9936238" />
